frontend/public/components/alert.tsx

```diff
@@ -0,0 +1,427 @@
+                    <dt>For</dt>
+                    <dd>{formatDuration(duration * 1000)}</dd>
+                  </React.Fragment>}
+                  <dt>Expression</dt>
+                  <dd><pre className="monitoring-query">{_.get(rule, 'query')}</pre></dd>
+                  {annotations.runbook_url && <React.Fragment>
+                    <dt>Runbook</dt>
+                    <dd><ExternalLink href={annotations.runbook_url} text={annotations.runbook_url} /></dd>
+                  </React.Fragment>}
+                </dl>
+              </div>
+            </div>
+          </div>
+        </div>
+        <div className="co-m-pane__body">
+          <div className="co-m-pane__body-group">
+            <SectionHeading text="Active Alerts" />
+            <div className="row">
+              <div className="col-xs-12">
+                {_.isEmpty(alerts) ? <div className="text-center">None Found</div> : <ActiveAlerts alerts={alerts} />}
+              </div>
+            </div>
+          </div>
+        </div>
+      </StatusBox>
+    </React.Fragment>;
+  }
+}
+export const AlertRulesDetailsPage = connectPage(connect(ruleStateToProps)(AlertRulesDetailsPage_));
+
+const Row = ({obj}) => {
+  const {alerts, annotations, labels, name} = obj;
+  const activeAt = _.min(_.map(alerts, 'activeAt'));
+
+  return <ResourceRow obj={obj}>
+    <div className="col-xs-7">
+      <div className="co-resource-link-wrapper">
+        <span className="co-resource-link">
+          <ResourceIcon kind={alertKind} />
+          <Link to={detailsURL('Alert', name, labels)} className="co-resource-link__resource-name">{name}</Link>
+        </span>
+      </div>
+      <div className="monitoring-description">{_.get(annotations, 'description') || _.get(annotations, 'message')}</div>
+    </div>
+    <div className="col-xs-3">
+      <State state={alertRuleState(obj)} />
+      {activeAt && <div className="text-muted monitoring-timestamp">since&nbsp;<Timestamp timestamp={activeAt} /></div>}
+    </div>
+    <div className="col-xs-2">{_.startCase(_.get(labels, 'severity', '-'))}</div>
+  </ResourceRow>;
+};
+
+const AlertHeader = props => <ListHeader>
+  <ColHead {...props} className="col-xs-7" sortField="name">Name</ColHead>
+  <ColHead {...props} className="col-xs-3" sortFunc="alertRuleState">State</ColHead>
+  <ColHead {...props} className="col-xs-2" sortField="labels.severity">Severity</ColHead>
+</ListHeader>;
+
+const rowFilter = {
+  type: 'alert-rule-state',
+  selected: ['firing', 'pending'],
+  reducer: alertRuleState,
+  items: [
+    {id: 'firing', title: 'Firing'},
+    {id: 'pending', title: 'Pending'},
+    {id: 'inactive', title: 'Inactive'},
+  ],
+};
+
+const nameFilterID = 'alert-rule-name';
+
+// Row filter settings are stored in "k8s"
+const listStateToProps = ({k8s}) => {
+  const filtersMap = k8s.getIn([reduxID, 'filters']);
+  return {filters: filtersMap ? filtersMap.toJS() : null};
+};
+
+class AlertsPage_ extends AlertsPageWrapper {
```

Comment:
Skeptical of the need to use inheritance, which is usually discouraged in React because it muddies the lifecycle hooks. Prefer composition instead.